flink-cdc-common/src/main/java/com/ververica/cdc/common/schema/Schema.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.common.schema;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import com.ververica.cdc.common.types.DataField;
+import com.ververica.cdc.common.types.DataType;
+import com.ververica.cdc.common.types.DataTypeRoot;
+import com.ververica.cdc.common.types.DataTypes;
+import com.ververica.cdc.common.types.RowType;
+import com.ververica.cdc.common.utils.Preconditions;
+
+import javax.annotation.Nullable;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.ververica.cdc.common.types.DataTypes.ROW;
+
+/** Schema of a table or data collection. */
+@PublicEvolving
+public class Schema implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private final Map<String, Column> columns;
+
+    private final List<String> primaryKeys;
+
+    private final Map<String, String> options;
+
+    private final @Nullable String comment;
+
+    private Schema(
+            Map<String, Column> columns,
+            List<String> primaryKeys,
+            Map<String, String> options,
+            @Nullable String comment) {
+        this.columns = columns;
+        this.primaryKeys = primaryKeys;
+        this.options = options;
+        this.comment = comment;
+    }
+
+    /** Returns the number of columns of this schema. */
+    public int getColumnCount() {
+        return columns.size();
+    }
+
+    /** Returns all {@link Column}s of this schema. */
+    public List<Column> getColumns() {
+        return new ArrayList<>(columns.values());
```

Comment:
Hi @leonardBang, this method may return a different order with the order column was added, should we add a list to maintain the order? it may lead to inconsistent order when creating sink table.

Reply:
I will open a PR to address this one